src/common/SegmentedButton.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import { ButtonStyle, getThemeColor, getReverseThemeColor } from "./Button";
+import ActionLink from './ActionLink';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCertificate } from "@fortawesome/free-solid-svg-icons";
+import Color from 'color';
+import classnames from "classnames";
+
+const SegmentedButton: React.FC<SegmentedButtonProps> = (props) => {
+  const [selected, setSelected] = useState(props.selected);
+  const THEME_COLOR = getThemeColor(props.theme);
+  const REVERSE_THEME_COLOR = getReverseThemeColor(props.theme);
+  const transparent = ButtonStyle.TRANSPARENT == props.theme;
+  return (
+    <div className={classnames("segmented-button", {})}>
+      {props.options.map((option) => (
+        <div className={classnames("segmented-button-option", selected == option.id ? "selected" : {})} key={"" + option.id}>
+          {option.updates && (
+            <div className="updates">
+              {option.updates === true ? (
+                <FontAwesomeIcon icon={faCertificate} />
+              ) : option.updates == Infinity ? (
+                "∞"
+              ) : (
+                option.updates
+              )}
+            </div>
+          )}
+          <ActionLink
+            link={{ 
+              onClick: () => {
+                setSelected(option.id);
+                if (option.onClick) {
+                  option.onClick();
+                }
+              },
+              href: option.href,
+            }}
+            className="segmented-button-link"
+            allowDefault={!!option.href && !option.onClick}
+          >
+            {option.label}
+          </ActionLink>
+        </div>
+      ))}
+      <style jsx>{`
+        .segmented-button {
+          border-radius: 25px;
+          background-color: ${REVERSE_THEME_COLOR};
+          border: 2px solid ${props.color || THEME_COLOR};
+          display: flex;
+          justify-content: space-evenly;
+        }
+        .segmented-button-option {
+          display: inline-block;
+          position: relative;
+        }
+        .segmented-button-option > :global(button),
+        .segmented-button-option > :global(a) {
+          border-radius: 25px;
+          padding: 8px 12px;
+          background-image: none;
+          color: ${props.color || THEME_COLOR};
+          background-color: ${REVERSE_THEME_COLOR};
+          border: 2px solid ${REVERSE_THEME_COLOR};
+          transition: all 0.2s linear 0s;
+          box-sizing: border-box;
+          display: inline-block;
+        }
+        .segmented-button-option > :global(button):focus,
+        .segmented-button-option > :global(a):focus {
+          outline: none;
+        }
+        .segmented-button-option:not(.selected):hover > :global(button),
+        .segmented-button-option:not(.selected):hover > :global(a) {
+          background-color: ${Color(props.color || THEME_COLOR).alpha(0.5)};
```

Comment:
Rather than using alpha, which has different effects according to the background the button is positioned in, we can change the brightness of the color using the Color library. Here's an example: https://github.com/essential-randomness/bobaboard-ui/blob/5d50ffe47497ad96f1c8fb248caf8aa9a3dd0e2c/src/board/BoardMenuItem.tsx#L36

You can play around with the numbers to see what looks good. I would start with 0.3.

Reply:
Ah, okay! My thinking was that the actual background of a `.segmented-button` would always be solid `REVERSE_THEME_COLOR`, so using `alpha` would always put the button pill halfway between `THEME_COLOR` and `REVERSE_THEME_COLOR`. But I agree it makes sense for consistency to just use `darken`.

Reply:
I notice that this has some perhaps undesired behavior with custom colors. 

#### `alpha(0.5)`
<img width="221" alt="Screen Shot 2021-03-04 at 3 46 35 PM" src="https://user-images.githubusercontent.com/7234006/110028350-1ffc4d80-7d01-11eb-8172-54f763331251.png">
<img width="216" alt="Screen Shot 2021-03-04 at 3 46 33 PM" src="https://user-images.githubusercontent.com/7234006/110028349-1ffc4d80-7d01-11eb-8b91-c64b761f3bca.png">

#### `darken(0.3)`
<img width="219" alt="Screen Shot 2021-03-04 at 3 46 17 PM" src="https://user-images.githubusercontent.com/7234006/110028347-1ffc4d80-7d01-11eb-8a8e-1cb6b79b4a21.png">
<img width="218" alt="Screen Shot 2021-03-04 at 3 46 13 PM" src="https://user-images.githubusercontent.com/7234006/110028344-1f63b700-7d01-11eb-8ab9-587a40fbe712.png">

#### `darken(0.5)`
<img width="216" alt="Screen Shot 2021-03-04 at 3 46 06 PM" src="https://user-images.githubusercontent.com/7234006/110028346-1ffc4d80-7d01-11eb-8aa8-73a64b7f9cbb.png">
<img width="219" alt="Screen Shot 2021-03-04 at 3 46 03 PM" src="https://user-images.githubusercontent.com/7234006/110028342-1f63b700-7d01-11eb-9939-eeab40e01cf7.png">

Reply:
Huh, interesting. I think you're right, it's better to go with the alpha at least for now. I need to figure out whether there's a better property than darkening for this. Thank you for trying~